pennylane/ops/qubit.py

```diff
@@ -265,6 +277,9 @@ def decomposition(wires):
         decomp_ops = [PhaseShift(np.pi / 2, wires=wires)]
         return decomp_ops
 
+    def adjoint(self, do_queue=False):
+        return PhaseShift(cmath.exp(-1j * np.pi / 2.0), wires=self.wires, do_queue=do_queue)
```

Comment:
Question: should we instead return
```suggestion
    def adjoint(self, do_queue=False):
        return S(wires=self.wires, do_queue=do_queue).inv()
```

My train of thought is that some frameworks might have different pathways for implementing `Sdg` as opposed to the phase shift gate.

Reply:
I really wanted to avoid using the inplace modifications. My main reasoning is to try to support `qml.adjoint(qml.adjoint(op))` and have it always work no matter what. Right now, there is no _clean_ way to do it use the existing `inv` modifier.

Reply:
In this case though, the suggestion isn't in-place? Since it is returning a new object `S` rather than `self`.

Reply:
Good point. I think for now that would be ok, but in the future I would prefer this method fully replace the `inv` op. That way, devices no longer need to deal with it themselves.

Reply:
Oh, one thing to note is that nested `adjoint`s (`adjoint(adjoint(SX))`) transforms won't work with gates that are defined by `inv`.